proxy/http2/HPACK.h

```diff
@@ -153,9 +165,9 @@ int64_t decode_string(Arena &arena, char **str, uint32_t &str_length, const uint
 
 int64_t encode_indexed_header_field(uint8_t *buf_start, const uint8_t *buf_end, uint32_t index);
 int64_t encode_literal_header_field(uint8_t *buf_start, const uint8_t *buf_end, const MIMEFieldWrapper &header, uint32_t index,
-                                    HpackFieldType type);
+                                    Http2DynamicTable &dynamic_table, HpackFieldType type);
 int64_t encode_literal_header_field(uint8_t *buf_start, const uint8_t *buf_end, const MIMEFieldWrapper &header,
```

Comment:
Those `encode_literal_header_field()` should be something like `encode_literal_header_field_with_indexed_name()` and `encode_literal_header_field_with_new_name()` ?